api.go

```diff
@@ -30,6 +30,42 @@ const (
 	)
 )
 
+func (h *Headscale) HealthHandler(
+	writer http.ResponseWriter,
+	req *http.Request,
+) {
+	respond := func(err error) {
+		writer.Header().Set("Content-Type", "application/json; charset=utf-8")
+
+		res := struct {
+			Healthy string `json:"healthy"`
+			Error   string `json:"error"`
+		}{
+			Healthy: "ok",
+		}
+
+		if err != nil {
+			writer.WriteHeader(http.StatusInternalServerError)
+			res.Error = err.Error()
+			res.Healthy = "error"
+		}
+
+		buf, err := json.Marshal(res)
+		if err != nil {
+			log.Error().Caller().Err(err).Msg("marshal failed")
+		}
+		writer.Write(buf)
+	}
+
+	err := h.pingDB()
+	if err != nil {
+		respond(err)
```

Comment:
We should not respond the error to the user, it is consider bad practice by OWASP;

https://cheatsheetseries.owasp.org/cheatsheets/Error_Handling_Cheat_Sheet.html

Reply:
ok will change that